bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/FlatRowAdapter.java

```diff
@@ -122,13 +122,23 @@ public void finishCell() {
     @Override
     public FlatRow finishRow() {
       ImmutableList.Builder<FlatRow.Cell> combined = ImmutableList.builderWithExpectedSize(totalCellCount);
+      FlatRow.Cell previous = null;
       for (List<FlatRow.Cell> familyCellList : cells.values()) {
-        combined.addAll(familyCellList);
+        for(FlatRow.Cell c : familyCellList) {
+          if(previous == null || !c.getLabels().isEmpty() || !keysMatch(c, previous)){
+            combined.add(c);
+          }
+          previous = c;
+        }
       }
 
       return new FlatRow(this.currentKey, combined.build());
     }
 
+    private boolean keysMatch(Cell c, Cell previous) {
+      return c.equals(previous);
```

Comment:
@sduskis 
I had a doubt here, Using Cell#equals would be ok? Or shall I compare specific field i.e. labels only?

Reply:
Please only use the qualifier and timestamp, since the row key and family are definitely going to be the same.

Reply:
I have updated to compare with qualifier & timestamp... but it seems labels comparison are needed for the case of `WhileMatchFilter`.